v22.1/row-level-ttl.md

```diff
@@ -0,0 +1,342 @@
+| `ttl_row_stats_poll_interval`                            | If set, counts rows and expired rows on the table to report as Prometheus metrics while the TTL job is running. Unset by default, meaning no stats are fetched and reported. | N/A                                 |
+| `ttl_pause` <a name="param-ttl-pause"></a>               | If set, stops the TTL job from executing.                                                                                                                                    | N/A                                 |
+| `ttl_job_cron` <a name="param-ttl-job-cron"></a>         | Frequency at which the TTL job runs, specified using [CRON syntax](https://cron.help).                                                                                       | N/A                                 |
+| `ttl_automatic_column`                                   | If set, use the value of the `crdb_internal_expiration` hidden column.                                                                                                       | N/A                                 |
+
+For more information about TTL-related cluster settings, see [View TTL-related cluster settings](#view-ttl-related-cluster-settings).
+
+## Examples
+
+### Create a table with row-level TTL
+
+To specify a TTL when creating a table, use the SQL syntax shown below. For example, to create a new table with rows that expire after 15 minutes, execute a statement like the following:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE events (
+  id UUID PRIMARY KEY default gen_random_uuid(),
+  description TEXT,
+  inserted_at TIMESTAMP default current_timestamp()
+) WITH (ttl_expire_after = '3 minutes');
+~~~
+
+~~~
+CREATE TABLE
+~~~
+
+Insert some data; it should work as expected:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+INSERT INTO events (description) VALUES ('a thing'), ('another thing'), ('yet another thing');
+~~~
+
+~~~
+INSERT 3
+~~~
+
+### Add row-level TTL to an existing table
+
+To add a TTL to an existing table, use the SQL syntax shown below.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER TABLE events SET (ttl_expire_after = '90 days');
+~~~
+
+~~~
+ALTER TABLE
+~~~
+
+### View TTL jobs
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT * FROM [SHOW SCHEDULES] WHERE label LIKE 'row-level-ttl-%';
+~~~
+
+~~~
+          id         |       label       | schedule_status |        next_run        | state | recurrence | jobsrunning | owner |            created            |     command
+---------------------+-------------------+-----------------+------------------------+-------+------------+-------------+-------+-------------------------------+-------------------
+  745396332264194049 | row-level-ttl-114 | ACTIVE          | 2022-03-17 21:00:00+00 | NULL  | @hourly    |           0 | root  | 2022-03-17 20:01:48.033267+00 | {"tableId": 114}
+(1 row)
+~~~
```

Comment:
maybe add a `SHOW JOBS` when the TTL job kicks off to show running jobs as well?

Reply:
Sorry, I don't understand.  I typed in `SHOW JOBS` but only saw the schema change job about changing the TTL storage param

`SHOW JOBS`

```
        job_id       |   job_type    |                                                        description                                                         | statement | user_name |  status   | running_status |          created           |          started           |          finished          |          modified          | fraction_completed | error | coordinator_id |      trace_id       |          last_run          |          next_run          | num_runs | execution_errors
---------------------+---------------+----------------------------------------------------------------------------------------------------------------------------+-----------+-----------+-----------+----------------+----------------------------+----------------------------+----------------------------+----------------------------+--------------------+-------+----------------+---------------------+----------------------------+----------------------------+----------+-------------------
  750771143879491585 | SCHEMA CHANGE | ALTER TABLE test.public.events SET (ttl_expire_after = '90 days')                                                          |           | root      | succeeded | NULL           | 2022-04-05 19:39:30.317168 | 2022-04-05 19:39:30.448445 | 2022-04-05 19:39:30.458783 | 2022-04-05 19:39:30.457604 |                  1 |       |              1 | 4535995618078903066 | 2022-04-05 19:39:30.448446 | 2022-04-05 19:40:00.448446 |        1 | {}
```

Reply:
`SHOW JOBS` will show the job when the TTL starts running. you may need to set `ttl_job_cron` to `* * * * *` and then wait a minute or two to show up.